src/Components/firebase-access.jsx

```diff
@@ -3,465 +3,386 @@ import "firebase/auth";
-      const currentPrice = symbolPricesData.prices[dayIndex];
-      prices.push(currentPrice);
-  }
+    changeCash(roomID,userID,dayIndex,changeArray,prices);
+    changeShares(roomID,userID,dayIndex,changeArray);
 
-  return prices;
+    return true;
 }
 
-export const getDayIndex = async (roomID) => {
-  const roomData = await getRoomData(roomID);
-  return roomData.day_index;
+function randomDate(start, end) {
+    var date = new Date(+start + Math.random() * (end - start));
+    return date;
 }
 
+export const setUpRoom = (NumOfSymbols,Rounds,userID,password,startingMoney = 10000) => {
 
-export const getDate = async (db, roomID) => {
-  const roomDoc = await db.collection('Rooms').doc(roomID).get();
-  const roomData = roomDoc.data();
-  return roomData.dates[roomData.day_index];
+    const roomRef = db.collection('Rooms').doc();
+    roomRef.set({
+        day_index: 0,
+        phase: 'no-host',
+        password: password,
+        starting_money: startingMoney,
+    });
+    const roomID = roomRef.id;
+    initSymbols(null,null,NumOfSymbols).then((symbolsL) => {
+        initDates(symbolsL,Rounds).then((datesD)=> {
+            roomRef.update({
+                symbols: symbolsL,
+                dates: datesD["dates"],
+            });
+            initializeQuiz(symbolsL,roomID,datesD["period"],datesD["dates"]);
+        });
+    });
+    return roomID;
 }
 
-// returns current price for all symbols being tracked
-export const getCurrentPrice = async (db, roomID) => {
-  const roomDoc = await db.collection('Rooms').doc(roomID).get();
-  const roomData = roomDoc.data();
-  const dayIndex = roomData.day_index;
-  const symbolNamesArray = roomData.symbols;
-
-  var prices = [];
-  
-  // save all current prices in an array
-  for (var index = 0; index < symbolNamesArray.length; index++) {
-    const currentName = symbolNamesArray[index];
-    const symbolPricesDoc = await db.collection('Rooms').doc(roomID).collection(currentName).doc('Prices').get();
-    const symbolPricesData = symbolPricesDoc.data();
-    
-    const currentPrice = symbolPricesData.prices[dayIndex];
-    prices.push(currentPrice);
-  }
-
-  return prices;
-}
-function compDoc(a, b){
-  if ( a.value < b.value ) {
-      return -1;
-  }
-  if ( a.value > b.value ) {
-    return 1;
-  }
-  return 0;
+//TODO: this method can be called many times, which leads to latency due to a lot of awaits. See if can pass userData from other methods
+export const updateNetWorth = async (roomID, userID) => {
+    const dayIndex = await getDayIndex(roomID);
+    const prices = await getPrices(roomID, dayIndex);
+    const userData = await getUserData(roomID, userID);
+    const userRef = getUserRef(roomID, userID);
+    const curShares = userData.curShares;
+    var netWorth = userData.money_left;
+    curShares.map((numShares,idx) => {netWorth += numShares * prices[idx];});
+    if (userData.net_worth == netWorth) return;
+    await userRef.update({net_worth: netWorth});
```

Comment:
don't need to await on update. documentation: https://firebase.google.com/docs/firestore/manage-data/add-data#web_8

```suggestion
    userRef.update({net_worth: netWorth});
```

Reply:
This is to make sure that a fetch function is not called before it finishes updating to avoid repetition